plugin/src/main/java/io/jenkins/plugins/analysis/core/util/ModelValidation.java

```diff
@@ -32,11 +32,10 @@
 @SuppressWarnings("PMD.GodClass")
 public class ModelValidation {
     private static final Set<String> ALL_CHARSETS = Charset.availableCharsets().keySet();
-    private static final Pattern VALID_ID_PATTERN = Pattern.compile("\\p{Alnum}[\\p{Alnum}-_]*");
+    private static final Pattern VALID_ID_PATTERN = Pattern.compile("\\p{Alnum}[\\p{Alnum}-_.]*");
```

Comment:
OK, this still starts with `\p{Alnum}`, so things like `.` or  `..` won't match. The pattern doesn't allow colons, slashes, backslashes, nor percent signs either. The pattern does not contain `^` and `$` anchors but it's used with [Matcher.matches()](https://docs.oracle.com/javase/8/docs/api/java/util/regex/Matcher.html#matches--) so it's implicitly anchored at both ends.

Reply:
I think it still is safe but there are enough characters available. Or are you missing some?

Reply:
No, I'm not missing any.